src/txn/txn_ckpt.c

```diff
@@ -994,8 +1076,17 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
 	if (full) {
 		__checkpoint_stats(session);
 #ifdef HAVE_TIMESTAMPS
-		__wt_timestamp_set(
-		    &conn->txn_global.last_ckpt_timestamp, &ckpt_tmp_ts);
+		/*
+		 * If timestamps were used to define the content of the
+		 * checkpoint update the saved last checkpoint timestamp,
+		 * otherwise leave it alone. If a checkpoint is taken without
+		 * timestamps, it's likely a bug, but we don't want to clear
+		 * the saved last checkpoint timestamp regardless.
+		 */
+		if (use_timestamp)
```

Comment:
This part of the change is causing a test failure - and I only added it out of paranoia - I'm going to back it out for now.

Reply:
I take that back - I just hadn't initialized use_timestamp..

Reply:
Any update on this part of the change?
I didn't understand why it is a bug to take a checkpoint without timestamps. I thought it is legal to do so even in a timestamp build. I understand ckpt_tmp_ts will be set to transaction's read timestamp. If a checkpoint is without a timestamp, would read timestamp of the transaction be set to 0, if yes only in that case we were clearing the last_ckpt_timestamp.